server/src/api/FriendRoutes.ts

```diff
@@ -1,26 +1,132 @@
 import express from "express";
 import { authenticateToken } from "../middlewares";
 import { prisma } from "../lib/prisma";
-import { OK_CODE } from "../lib/StatusCodes";
+import {
+  OK_CODE,
+  INTERNAL_ERROR_CODE,
+  NOT_FOUND_CODE,
+  INVALID_REQUEST_CODE,
+} from "../lib/StatusCodes";
 
 const router = express.Router();
 
-router.get("", authenticateToken, async (req, res) => {
-  const userId = req.body.user.id;
+router.get("/:id", authenticateToken, async (req, res) => {
+  const userId = Number(req.params.id);
 
-  const friends = await prisma.friend.findMany({
-    where: {
-      OR: [{ source_friend_id: userId }, { destination_friend_id: userId }],
-    },
-    include: { source_friend: true, destination_friend: true },
-  });
+  if (!userId) {
+    return res.sendStatus(INVALID_REQUEST_CODE);
+  }
 
-  let friendList = friends
-    .map((x) => x.source_friend)
-    .concat(friends.map((x) => x.destination_friend))
-    .filter((x) => x.id !== userId);
+  try {
+    const friends = await prisma.friend.findMany({
+      where: {
+        OR: [{ source_friend_id: userId }, { destination_friend_id: userId }],
+        is_accepted: true,
+      },
+      include: { source_friend: true, destination_friend: true },
+    });
 
-  res.status(OK_CODE).json(friendList);
+    let friendList = friends
+      .map((x) => x.source_friend)
+      .concat(friends.map((x) => x.destination_friend))
+      .filter((x) => x.id !== userId);
+
+    return res.status(OK_CODE).json(friendList);
+  } catch (error) {
+    return res.sendStatus(INTERNAL_ERROR_CODE);
+  }
+});
+
+router.get("/:id/requests", authenticateToken, async (req, res) => {
+  const userId = Number(req.params.id);
```

Comment:
also why is this from params? you have the user in all of these requests because the token is authenticated

Reply:
resolved to body.user.id for all